package/MDAnalysis/lib/picklable_file_io.py

```diff
@@ -0,0 +1,246 @@
+
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Picklable read-only I/O classes --- :mod:`MDAnalysis.lib.picklable_file_io`
+===========================================================================
+
+Provide with an interface for pickling read-only IO file object.
+These classes are used for further pickling :class:`MDAnalysis.core.universe`
+in a object composition approach.
+
+.. autoclass:: FileIOPicklable
+   :members:
+
+.. autoclass:: BufferIOPicklable
+   :members:
+
+.. autoclass:: TextIOPicklable
+   :members:
+
+.. autofunction:: pickle_open
+
+
+.. versionadded:: 2.0.0
+"""
+import io
+import os
+
+
+class FileIOPicklable(io.FileIO):
+    """File object (read-only) that can be pickled.
+
+    This class provides a file-like object (as returned by :func:`open`,
+    namely :class:`io.FileIO`) that, unlike standard Python file objects,
+    can be pickled. Only read mode is supported.
+
+    When the file is pickled, filename and position of the open file handle in
+    the file are saved. On unpickling, the file is opened by filename,
+    and the file is seeked to the saved position.
+    This means that for a successful unpickle, the original file still has to
+    be accessible with its filename.
+
+    Parameters
+    ----------
+    name : str
+        a filename given a text or byte string.
+
+    Example
+    -------
+    ::
+
+        >>> file = FileIOPicklable(PDB)
+        >>> file.readline()
+        >>> file_pickled = pickle.loads(pickle.dumps(file))
+        >>> print(file.tell(), file_pickled.tell())
+            55 55
+
+    See Also
+    ---------
+    TextIOPicklable
+    BufferIOPicklable
+
+
+    .. versionadded:: 2.0.0
+    """
+    def __init__(self, name):
+        super().__init__(name, mode='r')
```

Comment:
I think this and the following is what @kain88-de suggests, which could be easily added
```suggestion
    def __init__(self, name, mode='r'):
        self._mode = mode
        super().__init__(name, mode=mode)
```
I'm adding this for discussion, not for immediate inclusion.